src/simplewallet/simplewallet.cpp

```diff
@@ -6327,20 +6337,183 @@ bool simple_wallet::show_transfers(const std::vector<std::string> &args_)
       const tools::wallet2::unconfirmed_transfer_details &pd = i->second;
       uint64_t amount = pd.m_amount_in;
       uint64_t fee = amount - pd.m_amount_out;
+      std::vector<std::pair<std::string, uint64_t>> destinations;
+      for (const auto &d: pd.m_dests) {
+        destinations.push_back({get_account_address_as_str(m_wallet->nettype(), d.is_subaddress, d.addr), d.amount});
+      }
       std::string payment_id = string_tools::pod_to_hex(i->second.m_payment_id);
       if (payment_id.substr(16).find_first_not_of('0') == std::string::npos)
         payment_id = payment_id.substr(0,16);
       std::string note = m_wallet->get_tx_note(i->first);
       bool is_failed = pd.m_state == tools::wallet2::unconfirmed_transfer_details::failed;
       if ((failed && is_failed) || (!is_failed && pending)) {
-        message_writer() << (boost::format("%8.8s %6.6s %16.16s %20.20s %s %s %14.14s %s - %s") % (is_failed ? tr("failed") : tr("pending")) % tr("out") % get_human_readable_timestamp(pd.m_timestamp) % print_money(amount - pd.m_change - fee) % string_tools::pod_to_hex(i->first) % payment_id % print_money(fee) % print_subaddr_indices(pd.m_subaddr_indices) % note).str();
+        transfers.push_back({
+          .block = (is_failed ? tr("failed") : tr("pending")),
+          .timestamp = pd.m_timestamp,
+          .direction = "out",
+          .confirmed = false,
+          .amount = amount - pd.m_change - fee,
+          .hash = i->first,
+          .payment_id = payment_id,
+          .fee = fee,
+          .outputs = destinations,
+          .index = pd.m_subaddr_indices,
+          .note = note
+        });
+      }
+    }
+  }
+  // sort by block, then by timestamp (unconfirmed last)
+  std::sort(transfers.begin(), transfers.end(), [](const transfer_view& a, const transfer_view& b) -> bool {
+    if (a.confirmed && !b.confirmed)
+      return true;
+    if (a.block == b.block)
+      return a.timestamp < b.timestamp;
+    return a.block < b.block;
```

Comment:
Sort order for unconfirmed transactions may differ depending on the translation, not sure if you wanted a strict ordering for failed/pending/pool... probably doesn't matter

Reply:
![](https://media.giphy.com/media/FbZYCW78PANyM/giphy.gif)